commit 5a369c3b3a835019e3bc82d9da6331c6fb50b40e (refs/changes/22/4822/2)
Author: Jan Wyszynski <jan.wyszynski@joyent.com>
Date:   2018-09-11T19:38:06+00:00 (1 year, 1 month ago)
    
    MANTA-3941 garbage-collector transformer needs to ignore requests to flush empty shark list
    Reviewed by: Jared Morrow <jm@joyent.com>
    Approved by: Jared Morrow <jm@joyent.com>

diff --git a/lib/delete_record_transformer.js b/lib/delete_record_transformer.js
index 2d6987f..eeebed6 100644
--- a/lib/delete_record_transformer.js
+++ b/lib/delete_record_transformer.js
@@ -358,6 +358,10 @@ _listen_for_records()
 				self.mt_moray_listener.emit('cleanup', key);
 				return;
 			}
+			if (sharks.length === 0) {
+				self.mt_log.info('Skipping no-op flush.');
+				return;
+			}
 			self._flush(sharks, function (ferr) {
 				if (ferr) {
 					self.mt_log.warn({
